spec.bs

```diff
@@ -494,6 +506,218 @@ The <dfn for=Navigator method>runAdAuction(|config|)</dfn> method steps are:
+1. Set |pendingConfig|'s [=fenced frame config/mapped url=]'s [=mapped url/value=] to
+   |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/url=].
+1. Let |adSize| be |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/size=].
+1. If |adSize| is not null:
+   1. Set |pendingConfig|'s [=fenced frame config/content size=] to a struct with the following members:
+     : [=content size/value=]
+     :: |adSize| TODO: Resolve screen-relative sizes and macros and cast this properly.
+
+     : [=content size/visibility=]
+     :: "<a for=visibility>`opaque`</a>"
+1. Set |pendingConfig|'s [=fenced frame config/interest group descriptor=]'s
+   [=interest group descriptor/value=] to a struct with the following members:
+     : owner
+     :: |winningBid|'s [=generated bid/interest group=]'s [=interest group/owner=]
+
+     : name
+     :: |winningBid|'s [=generated bid/interest group=]'s [=interest group/name=]
+1. Set |pendingConfig|'s [=fenced frame config/fenced frame reporting metadata=] to a struct with
+   the following members:
+    : [=fenced frame reporting metadata/value=]
+    :: If |auctionConfig|'s [=auction config/component auctions=] is [=list/empty=] (i.e., if
+       there was no component auction), then a struct with the following members:
+       : [=fenced frame reporting metadata/fenced frame reporting map=]
+       :: a [=map=] «[ "<code>buyer</code>" → «», "<code>seller</code>" → «»]»
+
+       : [=fenced frame reporting metadata/direct seller is seller=]
+       :: true
+
+       Otherwise (i.e., if there was a component auction), a struct with the following members:
+       : [=fenced frame reporting metadata/fenced frame reporting map=]
+       :: a [=map=] «[ "<code>buyer</code>" → «», "<code>seller</code>" → «»,
+          "<code>component-seller</code>" → «»]»
+
+       : [=fenced frame reporting metadata/direct seller is seller=]
+       :: false
+
+    : [=fenced frame reporting metadata/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+1. Set |pendingConfig|'s [=fenced frame config/on navigate callback=] to a closure of
+   [=asynchronously finish reporting=] partially evaluated with a reference to |pendingConfig|'s
+   [=fenced frame config/fenced frame reporting metadata=]'s
+   [=fenced frame reporting metadata/value=]'s
+   [=fenced frame reporting metadata/fenced frame reporting map=].
+1. Set |pendingConfig|'s [=fenced frame config/nested configs=]'s [=nested configs/value=] to
+   the result of running [=create nested configs=] with |winningBid|'s
+   [=generated bid/ad component descriptors=].
+1. Return |pendingConfig|.
+
+</div>
+
+<div algorithm="asynchronously finish reporting">
+
+To <dfn>asynchronously finish reporting</dfn> given a reference to a
+[=fencedframetype/fenced frame reporting map=] |reportingMap| and [=leading bid info=]
+|leadingBidInfo|:
+1. TODO: [=Finalize a reporting destination=] with |reportingMap|

```

Comment:
What is left TODO here?

Reply:
This TODO will be in this PR. I need to look in the leadingBidInfo and finalize each reporting destination as appropriate.